docs/howto/configure-authd.md

```diff
@@ -39,26 +39,80 @@ Finally, as the supported authentication mechanism is the device workflow, you n
 client_id = <CLIENT_ID>
 
 [users]
-# The directory where the home directory will be created for new users.
-# Existing users will keep their current directory.
-# The user home directory will be created in the format of {home_base_dir}/{username}
-# home_base_dir = /home
-
-# The username suffixes that are allowed to login via ssh without existing previously in the system.
-# The suffixes must be separated by commas.
-# ssh_allowed_suffixes = @example.com,@anotherexample.com
+## The directory where the home directory will be created for new users.
+## Existing users will keep their current directory.
+## The user home directory will be created in the format of {home_base_dir}/{username}
+#home_base_dir = /home
+
+## The username suffixes that are allowed to login via ssh without existing previously in the system.
+## The suffixes must be separated by commas.
+#ssh_allowed_suffixes = @example.com,@anotherexample.com
+
+## 'allowed_users' specifies the users who are permitted to log in after
+## successfully authenticating with the Identity Provider.
+## Values are separated by commas. Supported values:
+## - 'OWNER': Grants access to the user specified in the 'owner' option
+##        	(see below). This is the default.
+## - 'ALL': Grants access to all users who successfully authenticate
+##      	with the Identity Provider.
+## - <username>: Grants access to specific additional users
+##           	(e.g. user1@example.com).
+## Example: allowed_users = OWNER,user1@example.com,admin@example.com
+#allowed_users = OWNER
+
+## 'owner' specifies the user assigned the owner role. This user is
+## permitted to log in if 'OWNER' is included in the 'allowed_users'
+## option.
+##
+## If this option is left unset, the first user to successfully log in
+## via this broker will automatically be assigned the owner role. A
+## drop-in configuration file will be created in broker.conf.d/ to set
+## the 'owner' option.
+##
+## To disable automatic assignment, you can either:
+## 1. Explicitly set this option to an empty value (e.g. owner = "")
+## 2. Remove 'OWNER' from the 'allowed_users' option
+##
+## Example: owner = user2@example.com
+#owner =
 ```
 
+The broker configuration consists of 2 sections. The `oidc` section and the `users` section.
+
+### OIDC section
+
+The `oidc` section configures how the authd-broker will interract with the OIDC Provider.
+
 Replace `<ISSUER_ID>` and `<CLIENT_ID>` by the issuer ID and client ID retrieved from the MS Entra ID portal.
 
+### Users section
+
+The users section configures which users will be allowed to access the machine.
+
+By default the first person to login to the machine is automatically registered
+as the owner. If you wish to override this behavior then directly register
+someone as the owner by using the `owner` keyword:
+
+```
+owner = your@email.com
+```
+
+Or specify a list of allowed users, by using the `allowed_users` keyword:
+
+```
+allowed_users = [person1@email.com,person2@email.com]
+```
+
```

Comment:
I'm not sure this is clear.

This section states that to override the automatic registration of an owner you can:

(1) specify the OWNER with the `owner` keyword **OR** (2) specify a list of users in `allowed_users`

It's clear that (1) will specify that the owner will be your@email.com, and that this will then not be automatically overridden (desirable), but it's not clear in (2) who the owner will be or if/how an owner will be created.

Perhaps a one-line explanation could be added to clarify that option. I think it should also be made clear that we are effectively removing OWNER from the default list in (2).

I note also that the two options here are different to the recommendations in the config file comments, this section shows steps to:

(1) set an owner and prevent auto assign from overriding that owner
(2) set no owner and prevent auto assign from creating any owner

versus

```ini
## To disable automatic assignment, you can either:
## 1. Explicitly set this option to an empty value (e.g. owner = "")
## 2. Remove 'OWNER' from the 'allowed_users' option
```

The latter seems more self-consistent: we are not setting an owner in both cases. In addition, it makes explicit that the OWNER is removed from the allowed users list, rather than merely defining an allowed users list.

So there are three options on this page, including the config comments:

* set an explicit owner
* remove owner from allowed users list
* set owner to empty string

All of these override the auto assignment but they do not all seem equivalent in their behaviour. The options that are shown are different between the config and the doc.

We do not have to explain everything here, but I think we should try to be clear and consistent.

I may have misunderstood something: please let me know if so :smile:

Reply:
You are correct.

`owner=` and removing `OWNER` from the `allowed_users` has the same effect, I didn't mention the former in order to keep it simple (this is just a how-to after all). But you are right, it needs some work.

I will try to make it clearer.

Reply:
Yes, I don't think we need to include everything here, but we just need to make sure what is included is maximally clear.

In this case, for example, we could just add a line after your second config example, just briefly explaining what it does.

It shouldn't take too much more text.